### DISCUSSION TITLE
blog/what-is-the-probability-that-two-persons-have-the-same-initials/

### DISCUSSION THREAD

**Discussion by giscus** [AntoineSoetewey/statsandr#110] [General]
# blog/what-is-the-probability-that-two-persons-have-the-same-initials/

Learn how to use simulations and for loops in R to answer a probability question such as What is the probability that two persons have the same initials?

https://statsandr.com/blog/what-is-the-probability-that-two-persons-have-the-same-initials/

**Comment by technocrat** [comment]
There are fewer combinations of two LETTERS taken two at a time (325) than there are days in a non-leap year (365). Therefore, we expect that the probability of two persons sharing a pair of initials (assuming that all pairs are equally probable, which isn't the case in practice because very few people are named Xerxes Xanadu compared to people named Meagan Kelly) is greater than two persons sharing the same birthday. Indeed, on day 23, when two birthdays are likely to be shared equals 0.507, the probability of shared initials is 0.549.

A solution to the birthday problem can be extended to the initials problem.

``` r
have_same <- function(s,n) {
  sample_space = s
  probability  = 1
  for (i in 0:(n - 1)) {
    probability = probability * (sample_space - i) / sample_space
  }
  1 - probability
}

# Example: Calculate the probability for n = 23 people
# sharing a birthday
birthday <- 365
people   <- 23
have_same(birthday,people)
#> [1] 0.5072972

results <- vector(length = birthday)
for(i in 1:birthday) results[i] = have_same(birthday,i)
plot(1:birthday,results,"l")
```

![](https://i.imgur.com/twQjxx6.png)

``` r

# sample space--the possible combination of THREE-letter initials
initials    <- dim(combn(LETTERS,3))[2]
have_same(initials,people)
#> [1] 0.09297894

results <- vector(length = initials)
for(i in 1:initials) results[i] = have_same(initials,i)
plot(1:initials,results,"l")
```

![](https://i.imgur.com/eESuSbt.png)

<sup>Created on 2023-12-06 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Reply by AntoineSoetewey** [reply]
Dear @technocrat,

Thanks for your input!

You're right, all pairs of initials are assumed to be equally probable in this post, although in practice it is not the case. I didn't correct for this, as the goal was more to illustrate a process (i.e., estimating a probability through simulations). This point has been mentioned at the end of the post though.

Regarding the other point, there are indeed 325 combinations of two-letters initials, if we consider that repetitions are not allowed and the order does not matter. This means that:

- AA, BB, etc. are not allowed, and
- AB and BA are considered only one pair of initials.

If we follow these principles, I agree that we have 325 combinations, which can be computed in R with `choose(n = 26, k = 2)`.

However, in our case, repetitions are allowed and order does matter:

- someone could be called Oliver Olson (initials = OO), and
- someone called Meagan Kelly have different initials than someone called Kelly Moore (MK versus KM).

Following this, I believe there are $26^2 = 676$ possible pairs of initials and not 325. Am I missing something?

Moreover, as also pointed in the conclusion, I focused only on two-letters initials and not three-letters. In the second part of your code, you seem to use three-letters initials. For greater clarity, I added a sentence explaining this in the body of the post (just after the introduction) as well.

Now, if I adapt your code to my needs, I believe it is a very convenient way to compare results obtained through simulations. For this reason, I have added a section in the post which compares results obtained via the two methods. Many thanks for that! Do not hesitate to let me know if I misunderstood something.

Regards,
Antoine

**Reply by technocrat** [reply]
Hi, Antone

I blew that one, of course, *combinations* of birthdays but *permutations* of initials. I only used 3-letter initials because I also stupidly assume that 365 and 325 were too close. Here's corrected code. I suppose when life was more parochial we could have sampled names from a phone directory or similar source to put some shape on the probability of any initials pair being drawn. Our worlds are so far-flung anymore, that might be a challenge!

``` r
have_same <- function(s,n) {
  sample_space = s
  probability  = 1
  for (i in 0:(n - 1)) {
    probability = probability * (sample_space - i) / sample_space
  }
  1 - probability
}

# Example: Calculate the probability for n = 23 people
# sharing a birthday
birthday <- 365
people   <- 23
have_same(birthday,people)
#> [1] 0.5072972

results <- vector(length = birthday)
for(i in 1:birthday) results[i] = have_same(birthday,i)
plot(1:birthday,results,"l")
```

![](https://i.imgur.com/0jaZaRN.png)

``` r

# sample space--the possible PERMUTATIONS of TWO-letter initials
library(gtools)
initials    <- dim(permutations(n = 26, r = 2, v = letters))[1]
have_same(initials,people)
#> [1] 0.3255119

results <- vector(length = initials)
for(i in 1:initials) results[i] = have_same(initials,i)
plot(1:initials,results,"l")
```

![](https://i.imgur.com/2e2MyD3.png)

<sup>Created on 2023-12-07 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Reply by AntoineSoetewey** [reply]
Thanks!

Isn't there still a problem with `dim(permutations(n = 26, r = 2, v = letters))[1]`?

This gives 650, which is correct only if repetitions are not allowed. But as mentioned, it's possible to have initials with two times the same letter.

I simply used `26^2` in the code of the post.

**Reply by technocrat** [reply]
For the price of a small embarrassment, I now have forever knowledge of the difference! 26^2 is the case with duplication and 26^2 - 26 without!

**Reply by AntoineSoetewey** [reply]
No problem!

Thanks for your code, it was a great input to the post.
